protocol.md

```diff
@@ -114,6 +114,114 @@ Observe the behavior of this algorithm in some failure scenarios. First, any of
+
+The sharer installs software on their phone that manages information for them that they want to keep secret, but also want to back up. This information is called the \"secret\". The secret might be a single key. Or it might contain a list of passwords and a list of cryptographic keys and some combination lock combos and text describing a secret recipe. It is whatever information the app needs to back up to enable decentralized recovery.
+
+The software on the phone can use DeRec to back up the secret in a way that allows recovery if the phone is lost or destroyed. The software informs the user that this will happen if they connect to multiple \"helpers\", who are people or organizations willing to keep \"shares\" of that secret. A single share reveals nothing about the secret, so the helpers do not know what the user is backing up.
+
+The sharer\'s app generates a random number which is the \"secret ID\", which is used to identify this secret in the future. The user might choose to have more than one secret. For example, they might have a secret on the phone they normally carry, which is shared with only a few helpers. And then they have a higher-valued secret stored on a laptop that is kept in a safe, and is shared with a larger number of helpers. In that case, each secret would have its own random secret ID. If there is a helper that is acting as a helper for both secrets, then the user will have paired with them twice, and they will store both shares, each with its associated secret ID. Or, a sharer might have only a single secret, and each of their helpers will store that secret and that single secret ID.
+
+The user can pair with a helper in person, or it could be partially automated. The protocol assumes that one party gives the other contact information outside the protocol, which contains only the minimum information needed to start the protocol. The other party then sends a \"pairing request\", and receives back a \"pairing reply\". At that point, the two are paired.
+
+Partially automated pairing might happen between a customer and a company providing them an app. This might be a bank or credit union customer that gets the banking app from the bank. The customer might go through some kind of KYC (Know Your Customer) process where the bank verifies their identity before establishing an account for them. That process would end with them having an account and the app to access the account. It would also end up sending the app the contact information. and the user\'s app would then send the pairing request to the bank\'s server, which would reply. The contact information contains a random number which is the \"nonce\", which the user sends back to the bank as part of the pairing request. This allows the bank to ensure that the pairing request is coming from the same person who KYCed and established the account. In this scenario, the bank has servers that fill the role of helper.
```

Comment:
Should the nonce be authenticated, or at least made unmalleable?

Reply:
(where not in person)

Reply:
The initial nonce is transferred from responder to initiator out of band, outside the protocol, along with the public key. Then, the protocol defines two messages, one from initiator to responder, and the other the other direction. Both of those messages are signed and encrypted, so the nonce in them cannot be modified. And after those two messages, the nonce is never used again, and both parties can discard it.

The nonce is authenticated. In fact, it's basically the token that says authentication has taken place. In one scenario, Alice and Bob have apps on their phones, and they physically meet in the same room, and talk to each other and authenticate each other. Then Alice's phone camera looks at a QR code displayed on Bob's phone. Bob's phone will have created the nonce and put it in that QR code. If Alice then send him the first message of the protocol, and Bob sees that nonce inside that signed-and-encrypted message, then Bob can know for sure that he received that pairing message from someone who is looking at his phone screen at that moment.

Reply:
got it, thanks!